FlowCryptAppTests/Core/FlowCryptCoreTests.swift

```diff
@@ -370,12 +377,53 @@ final class FlowCryptCoreTests: XCTestCase {
         XCTAssertEqual(decrypted.content.count, fileData.count)
     }
 
-    func testException() throws {
+    func testException() async throws {
         do {
-            _ = try core.decryptKey(armoredPrv: "not really a key", passphrase: "whatnot")
+            _ = try await core.decryptKey(armoredPrv: "not really a key", passphrase: "whatnot")
             XCTFail("Should have thrown above")
         } catch let CoreError.exception(message) {
             XCTAssertNotNil(message.range(of: "Error: Misformed armored text"))
         }
     }
+
+    // this test is only meaningful on a real device
+    // it passes on simulator even if implementation is broken
+    // maybe there's a way to run simulator with more cores? (on a mac that has them) which would simulate real device better
+    func testCoreResponseCorrectnessUnderConcurrency() async throws {
```

Comment:
Is there a way to run this code on simulator that actually uses many OS threads for the simulator?